libraries/app/database_api.cpp

```diff
@@ -1308,34 +1308,30 @@ order_book database_api_impl::get_order_book( const string& base, const string&
    return result;
 }
 
-vector<market_volume> database_api::get_top_markets(uint32_t limit)const
+vector<market_ticker> database_api::get_top_markets(uint32_t limit)const
 {
    return my->get_top_markets(limit);
 }
 
-vector<market_volume> database_api_impl::get_top_markets(uint32_t limit)const
+vector<market_ticker> database_api_impl::get_top_markets(uint32_t limit)const
 {
    FC_ASSERT( _app_options && _app_options->has_market_history_plugin, "Market history plugin is not enabled." );
 
    FC_ASSERT( limit <= 100 );
 
    const auto& volume_idx = _db.get_index_type<graphene::market_history::market_ticker_index>().indices().get<by_volume>();
    auto itr = volume_idx.rbegin();
-   vector<market_volume> result;
+   vector<market_ticker> result;
    result.reserve(limit);
 
-   const fc::time_point_sec now = fc::time_point::now();
+   const fc::time_point_sec now = _db.head_block_time();
 
    while( itr != volume_idx.rend() && result.size() < limit)
    {
-      market_volume mv;
-      mv.time = now;
       const auto assets = get_assets( { itr->base, itr->quote } );
-      mv.base = assets[0]->symbol;
-      mv.quote = assets[1]->symbol;
-      mv.base_volume = uint128_amount_to_string( itr->base_volume, assets[0]->precision );
-      mv.quote_volume = uint128_amount_to_string( itr->quote_volume, assets[1]->precision );
-      result.emplace_back( std::move(mv) );
+      market_ticker mt = get_ticker( assets[0]->symbol, assets[1]->symbol, true );
```

Comment:
Calling `get_ticker` directly is inefficient, since it will query the database again for same data. We already have `itr` here so just need to get the data. I recommend that we move the code in `get_ticket` to a new static function and call it in both `get_ticker` and here.